modules/sw360/src/main/java/org/eclipse/sw360/antenna/sw360/workflow/generators/SW360Updater.java

```diff
@@ -58,22 +46,26 @@ public void configure(Map<String, String> configMap) throws AntennaConfiguration
         projectName = retrieveName(configuredSW360Project);
         projectVersion = retrieveVersion(configuredSW360Project);
 
-        sw360RestServerUrl = getConfigValue(REST_SERVER_URL_KEY, configMap);
-        sw360AuthServerUrl = getConfigValue(AUTH_SERVER_URL_KEY, configMap);
-        sw360User = getConfigValue(USERNAME_KEY, configMap);
-        sw360Password = getConfigValue(PASSWORD_KEY, configMap);
-        sw360ClientId = getConfigValue(CLIENT_USER_KEY, configMap);
-        sw360ClientPassword = getConfigValue(CLIENT_PASSWORD_KEY, configMap);
-        sw360ProxyUse = Boolean.parseBoolean(getConfigValue(PROXY_USE, configMap, "false"));
-        sw360ProxyHost = context.getToolConfiguration().getProxyHost();
-        sw360ProxyPort = context.getToolConfiguration().getProxyPort();
+        // Proxy configuration
+        String sw360ProxyHost = context.getToolConfiguration().getProxyHost();
+        int sw360ProxyPort = context.getToolConfiguration().getProxyPort();
+
+        SW360ConnectionConfiguration sw360ConnectionConfiguration = new SW360ConnectionConfiguration(key -> getConfigValue(key, configMap),
+                key -> getBooleanConfigValue(key, configMap),
+                sw360ProxyHost, sw360ProxyPort);
+
+        // General configuration
+        updateReleases = getBooleanConfigValue(UPDATE_RELEASES, configMap);
+        if (updateReleases) {
+            throw new AntennaExecutionException("The functionallity to update releases, activated with the " + UPDATE_RELEASES + " is not yet supported.");
+        }
+        Boolean uploadSources = getBooleanConfigValue(UPLOAD_SOURCES, configMap);
+
+        sw360MetaDataUpdater = new SW360MetaDataUpdater(sw360ConnectionConfiguration, uploadSources);
```

Comment:
I don't agree with this. 
You are giving a variable to the `SW360MetaDataUpdater` that are called "updateReleases" there. And hence will call `updateRelease()` you specifically mentioned not to be activated above. and no functionality of source uploading is activated in `updateRelease()` or `patchRelease()`.

`SW360MetaDataUpdater` should not instantiate the `updateRelease` in its constructor. You could rename the variable in the constructor and make a new field.

Reply:
sorry, the code is here not complete. I will complete it immediately.

Reply:
please look at https://github.com/eclipse/antenna/compare/85e302f6ee93fe2ffaea2de5da266f604530a155..fe5f8668b982dba68e2d8e5aaa800bb0ce9e4db9